File: app/src/main/java/org/tvheadend/tvhclient/data/source/DataSourceInterface.java
package org.tvheadend.tvhclient.data.source;

import android.arch.lifecycle.LiveData;

import java.util.List;

interface DataSourceInterface<T> {

    void addItem(T item);

    void updateItem(T item);

    void removeItem(T item);

    T getItemById(Object id);

    List<T> getItems();

    LiveData<Integer> getLiveDataItemCount();

    LiveData<List<T>> getLiveDataItems();

    LiveData<T> getLiveDataItemById(Object id);
}

File: app/src/main/java/org/tvheadend/tvhclient/data/source/DataSourceInterface.kt
package org.tvheadend.tvhclient.data.source

import androidx.lifecycle.LiveData

interface DataSourceInterface<T> {

    fun getLiveDataItemCount(): LiveData<Int>

    fun getLiveDataItems(): LiveData<List<T>>

    fun getLiveDataItemById(id: Any): LiveData<T>

    fun getItems(): List<T>

    fun getItemById(id: Any): T?

    fun addItem(item: T)

    fun updateItem(item: T)

    fun removeItem(item: T)
}
